runner/src/main.rs

```diff
@@ -252,28 +245,46 @@ fn build_server(opt: &BuildServer) -> Step {
 
 fn run_tests() -> Step {
     Step::Multiple {
-        name: "root".to_string(),
+        name: "tests".to_string(),
         steps: vec![
-            run_buildifier(),
-            run_prettier(),
-            run_embedmd(),
-            run_cargo_fmt(),
-            run_cargo_test(),
-            run_cargo_doc_test(),
+            run_buildifier(FormatMode::Check),
+            run_prettier(FormatMode::Check),
+            run_markdownlint(FormatMode::Check),
+            run_liche(),
+            run_cargo_fmt(FormatMode::Check),
             run_cargo_clippy(),
+            run_cargo_test(),
             run_bazel_build(),
             run_bazel_test(),
         ],
     }
 }
 
-/// Returns all the files and directories under the examples directory.
-fn examples() -> impl Iterator<Item = PathBuf> {
-    let walker = walkdir::WalkDir::new("./examples").into_iter();
-    walker
-        .filter_entry(|e| !is_ignored_entry(e))
-        .filter_map(Result::ok)
-        .map(|e| e.into_path())
+fn format() -> Step {
+    Step::Multiple {
+        name: "format".to_string(),
+        steps: vec![
+            run_buildifier(FormatMode::Fix),
+            run_prettier(FormatMode::Fix),
+            run_markdownlint(FormatMode::Fix),
+            run_embedmd(FormatMode::Fix),
+            run_cargo_fmt(FormatMode::Fix),
+        ],
+    }
+}
+
+fn run_ci() -> Step {
+    Step::Multiple {
+        name: "ci".to_string(),
+        steps: vec![
+            run_tests(),
+            run_examples(&RunExamples {
+                application_variant: "rust".to_string(),
+                example_name: None,
```

Comment:
If I understand Line 140 correctly, this means that all examples will be run?

Asking because as somebody not familiar with the code (and maybe just Rust), `example_name: None` reads a bit like "don't run any examples", but maybe that's just me. Perhaps could be improved by changing the naming to something like "only_run_example_name", but 🤷‍♀️

Reply:
Yeah I agree, added a TODO on the struct field itself for now.